cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaBrokerConfigurationBuilder.java

```diff
@@ -710,6 +716,7 @@ private void configureKeycloakAuthorization(String clusterName, KafkaAuthorizati
         addOption(writer, "strimzi.authorization.grants.refresh.pool.size", authorization.getGrantsRefreshPoolSize());
         addOption(writer, "strimzi.authorization.connect.timeout.seconds", authorization.getConnectTimeoutSeconds());
         addOption(writer, "strimzi.authorization.read.timeout.seconds", authorization.getReadTimeoutSeconds());
+        addOption(writer, "strimzi.authorization.http.retries", authorization.getHttpRetries());
```

Comment:
It is not really an issue ... more a curiosity ... why do the OAuth options have the null and >0 checks and the authorization options are just added? Also, what if this is `null` given the type seems to be an Integer?

Reply:
This is probably just an ommition on my part. It would make sense that checks for configuration that will result in failure if let though should be a reason for aborting at the operator level. I'll have a closer look at this.

Reply:
Just to be clear, if it works fine, I do not have problem with it ... I was just wondering as you have the checks few lines above.

Reply:
Oh I see what you mean. Yeah, this part is not necessarily consistent. 

The easiest would be to perform no checks and to let all values through without any validation.
The minus of that is that users can't look at operator log for configuration errors but have to look in Kafka log and decipher how oauth.* config properties convert back to listener / authorizer CRD config names.

So it's good to have validation at the Operator level. Then next thing is trying to decide between raising an error for every obviously nonsensical config value (which may sometimes be deliberately done during trying out different things), and interpreting a nonsensical value as something to silently ignore (and not set the value at all). The same choice happens inside OAuth when interpreting the configuration. It ignores some value (but logs a warning) and you always can see the actual configuration values used in the log. Of course it depends on the specific configuration option if it can be interpreted clearly. For example if only positive integer makes sense then do we raise an error for negative or zero value or just ignore it and use zero default.

For some options (those a few lines above) I opted in the past to ignore the bad value, letting the OAuth use the default.